torch/distributed/pipelining/stage.py

```diff
@@ -822,7 +822,8 @@ def backward_one_chunk(
             # to return to the user in merge_output_chunks, therefore
             # this should be detached to release autograd graph context and free memory earlier
             for t in stage_output:
-                t.detach_()
+                if not t._is_view():  # views are not detachable in-place
+                    t.detach_()
```

Comment:
hm. this change is probably strictly correct, but it looks like we need another way to release activation memory if the user model returns a view.

we can at least issue a warning here, and ask the user to release their own memory by detaching, or else avoid returning a view from forward.

what else could we do 🤔 
cc @H-Huang